apps/api/src/modules/job/job.service.ts

```diff
@@ -1,21 +1,43 @@
 import {
+  CreateJobDto,
   EditJobDto,
   GetJobCardWithMaxPageDto,
   JwtDto,
   SearchJobDto,
   UserType,
 } from '@modela/dtos'
 import { Injectable, NotFoundException } from '@nestjs/common'
-import { ForbiddenException } from '@nestjs/common/exceptions'
+import {
+  ForbiddenException,
+  InternalServerErrorException,
+} from '@nestjs/common/exceptions'
 
 import { JobRepository } from './job.repository'
 
 @Injectable()
 export class JobService {
   constructor(private repository: JobRepository) {}
 
-  create(createJobDto: EditJobDto) {
-    return 'This action adds a new job'
+  async createJob(createJobDto: CreateJobDto, userId: number) {
+    // if minAge is greater than maxAge, throw error
+    if (createJobDto.minAge > createJobDto.maxAge) {
+      throw new ForbiddenException('minAge is greater than maxAge')
+      return
+    }
+    // for each shooting, if the start time is greater than the end time, throw error
+    for (let i = 0; i < createJobDto.shooting.length; i++) {
+      if (
+        createJobDto.shooting[i].startTime > createJobDto.shooting[i].endTime
+      ) {
+        throw new ForbiddenException('startTime is greater than endTime')
+        return
+      }
+    }
+    try {
+      await this.repository.createJob(createJobDto, userId)
+    } catch (e) {
+      throw new InternalServerErrorException()
+    }
```

Comment:
Should do something with e(error) like console.log(e). If not there is no need to try catch